### DISCUSSION TITLE
Ending the current work.

### DISCUSSION THREAD

**Discussion by eminemirhansener** [igece/SoxSharp#29] [Q&A]
Hi!
I am trying to make a real-time bandpass filtering application. I created a thread to make sox function to work simultaneously with other processes but couldn't handle to end the current work by button click event.

Here is my sox function;
```
private void sox(int frequency, int width)
        {

            using (var sox = new Sox("C:\\Program Files (x86)\\sox-14-4-1\\sox.exe"))
            {
                sox.Effects.Add(new VolumeEffect(0, GainType.Db));
                sox.Effects.Add(new BandPassFilterEffect(frequency, width));
                sox.OnProgress += sox_OnProgress;
                sox.Process("-d", "-d");
            }

        }
```

I used boolean btn_clicked variable which is enabled by button, inside OnProgress event handler but it didn't work for me.

```
        void sox_OnProgress(object sender, ProgressEventArgs e)
        {
            if (btn_clckd)
                e.Abort = true;

            btn_clckd = false;
        }
```

Long story short, I would like to know if I make mistakes or how could make I it done?

Also the full project is in [my repository page](https://github.com/eminemirhansener/Realtime-Audio-Filtering-VS.net).

Thanks :)

**Comment by eminemirhansener** [comment]
Guess it was working but it is too delayed. Is there any better solution?

**Comment by igece** [comment]
When you say delayed, you mean it takes too long between you click the button and the operation is canceled?

**Reply by eminemirhansener** [reply]
Yes, it takes about 5 sec's. But it's not always same delay.

**Reply by igece** [reply]
Ok, I will take a look at it at soon as I have some time.
